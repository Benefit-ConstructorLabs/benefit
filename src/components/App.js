import React from 'react';
import AppHeader from './AppHeader';
import NewRecipientContainer from '../containers/NewRecipientContainer';
import Recipient from './Recipient';
import NewDonation from './NewDonation';
import AppFooter from './AppFooter';

import '../../styles/components/app.scss';

const App = () => (
  <React.Fragment>
    <p>The app...</p>
    <AppHeader />
    <NewRecipientContainer />
    <Recipient />
    <NewDonation />
    <AppFooter />
  </React.Fragment>
);

export default App;
