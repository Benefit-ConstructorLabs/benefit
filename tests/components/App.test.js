import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../src/components/App';


describe('App', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
