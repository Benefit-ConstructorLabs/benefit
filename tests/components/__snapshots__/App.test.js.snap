// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
Array [
  <p>
    The app...
  </p>,
  <h1>
    App Header — Benefit
  </h1>,
  <h2>
    New Recipient
  </h2>,
  <h2
    className="newrecipient__title"
  >
    Start taking digital donations in 3 steps
  </h2>,
  <img
    alt=""
    height=""
    src=""
    width=""
  />,
  <form>
    <label
      className="newrecipient__intro"
    >
      Add your basic information
    </label>
    <input
      placeholder="First name"
      text="firstname"
      type="text"
    />
    <input
      placeholder="Last name"
      text="lastname"
      type="text"
    />
    <input
      placeholder="Telephone number"
      text="tel"
      type="text"
    />
    <button
      className="newrecipient__upload-button"
    >
      Upload a picture
    </button>
    <label>
      Now tell people 3 things about you...
    </label>
    <input
      placeholder="1. I play the piano"
      text="text"
      type="text"
    />
    <input
      placeholder="2. I like white coffee"
      text="text"
      type="text"
    />
    <input
      placeholder="3. I'm a Rugby fan"
      text="text"
      type="text"
    />
    <button
      className="newrecipient__qr-button"
    >
      Generate Your Unique QR Code
    </button>
  </form>,
  <p>
    Recipient sign-up form
  </p>,
  <h2>
    Recipient Component
  </h2>,
  <h2>
    You Unique QR code
  </h2>,
  <canvas />,
  <h3
    className="recipient__steps"
  >
    3 easy steps to receive donations
  </h3>,
  <ul
    className="recipient__steps-list"
  >
    <li>
      Display your unique code to potential donors
    </li>
    <li>
      Donors scan your QR code and set a payment
    </li>
    <li>
      Check your total donations
    </li>
  </ul>,
  <button
    className="recipient__print-qrcode"
    type="button"
  >
    Print your QR code
  </button>,
  <button
    className="recipient__view-donations"
    type="button"
  >
    See your donation total
  </button>,
  <div
    className="donation"
  >
    <h2
      className="donation__headline"
    >
      Cashless donations that make a difference
    </h2>
    <p
      className="donation__strapline"
    >
      Quick. Secure. Cashless.
    </p>
    <ol
      className="donation__steps"
    >
      <li
        className="donation__step"
      >
        Decide how much you would like to donate to firstname.
      </li>
      <li
        className="donation__step"
      >
        Set your payment amount.
      </li>
      <li
        className="donation__step"
      >
        Help firsname out.
      </li>
    </ol>
  </div>,
  <div
    className="donation"
  >
    <img
      alt="Recipient"
      className="donation__recipient__photo"
      src="https://via.placeholder.com/150?text=Recipient+photo"
    />
    <h3
      className="donation__purpose"
    >
      Donate towards a night in a shelter
    </h3>
    <form
      className="donation__form"
      onSubmit={[Function]}
    >
      <h4>
        How much?
      </h4>
      <p>
        <input
          className="donation__form__amount"
          id="donation__amount"
          onChange={[Function]}
          placeholder="£undefined"
          type="number"
        />
      </p>
      <p>
        <input
          type="range"
        />
      </p>
      <p
        className="donation__form__quick-amount"
      >
        <button
          className="btn btn__quick-amount"
          type="button"
        >
          50p
        </button>
        <button
          className="btn btn__quick-amount"
          type="button"
        >
          £1
        </button>
        <button
          className="btn btn__quick-amount"
          type="button"
        >
          £2
        </button>
      </p>
      <button
        className="btn btn__submit"
        type="submit"
      >
        Donate to firstname
      </button>
    </form>
  </div>,
  <div
    className="recipient-profile"
  >
    <h3
      className="recipient-profile__headline"
    >
      Fancy a chat?
    </h3>
    <p
      className="recipient-profile__intro"
    >
      firstName is interested in:
    </p>
    <ol
      className="recipient-profile__interests"
    >
      <li
        className="recipient-profile__interest"
      >
        First interest
      </li>
      <li
        className="recipient-profile__interest"
      >
        Second interest
      </li>
      <li
        className="recipient-profile__interest"
      >
        Third interest
      </li>
    </ol>
  </div>,
  <h2>
    Donation Payment
  </h2>,
  <div
    className="payment-details-form"
  >
    <h3>
      Set up payment
    </h3>
    <form
      className="payment-details-form__form"
    >
      <input
        placeholder="Card number"
        type="text"
      />
      <input
        placeholder="Expiry date"
        type="text"
      />
      <input
        placeholder="CCV"
        type="text"
      />
      <p>
        Set up a donor account?
      </p>
      <p>
        Continue donating anonymously:
      </p>
      <button
        type="button"
      >
        Donate -Price- to -First Name-
      </button>
    </form>
  </div>,
  <section
    className="acknowledgement"
  >
    <p
      className="acknowledgement__notification"
    >
      Donation confirmed
    </p>
    <h2
      className="acknowledgement__confirmation"
    >
      All set. Thanks for helping “ FirstName” out!
    </h2>
    <h3
      className="acknowledgement__question"
    >
      Want to save your details for another donation?
    </h3>
    <button
      className="btn acknowledgement__account__btn"
      onClick={[Function]}
      type="button"
    >
      Set up donor account
    </button>
  </section>,
  <h1>
    App Footer — © 2018 Benefit Team
  </h1>,
]
`;
